### PR TITLE
Filter hadoop collector for whitelist/blacklist 

### PR DESCRIPTION
**Pull request by ethan-eb**
python-diamond/Diamond#234

This change adds a new path_allowed method that checks white/blacklists against full path instead of just the "name." This is desirable in hadoop because there are path variations for a given name that might need to be handled differently.

The new method is called from the hadoop collector, so it doesn't affect existing white/blacklist behaviors for other collectors.

In the test I added a 'metrics_blacklist' config, confirmed that broke the test, and then adjusted expected.pkl to correct the test.

### CODE REVIEW

File: src/collectors/hadoop/hadoop.py [hunk 1]
```diff
@@ -101,6 +101,10 @@ def collect_from(self, filename):
                             data['name'],
                             metric, ]))
 
+                    if not self.path_allowed(path):
+                        # whitelist / blacklist
+                        continue
+
                     value = float(metrics[metric])
 
                     self.publish_metric(Metric(path,
```
Comment (jaingaurav) [inline src/collectors/hadoop/hadoop.py]: Weren't you going to switch to the publish method?

Comment (ethan-eb) [inline src/collectors/hadoop/hadoop.py]: So I was trying to keep the issues and pull requests isolated, but there is overlap and so it is somewhat confusing. In my PR for multi black/white lists #245 I did move it to the publish method. Also renamed it metric_allowed.

I can combine these two things (path vs name & single vs multiple white/black lists) into one PR. Whatever you think makes more sense.

I'm going to rebase my existing PRs with the latest now...

**Comment by ebpitts** [comment]
If #245 is merged this can be closed, since it's a subset of that PR.

**Comment by shortdudey123** [comment]
Subset of https://github.com/python-diamond/Diamond/pull/245

**Pull request closed without merging**